synfig-studio/src/synfigapp/actions/waypointsetsmart.cpp

```diff
@@ -111,10 +116,12 @@ bool
 Action::WaypointSetSmart::is_candidate(const ParamList &x)
 {
 	return (candidate_check(get_param_vocab(),x) &&
-			// We need an animated valuenode.
-			ValueNode_Animated::Handle::cast_dynamic(x.find("value_node")->second.get_value_node()) &&
 			// We need either a waypoint or a time.
-			(x.count("waypoint") || x.count("time")));
+			(x.count("waypoint") || x.count("time")) &&
+			// Parameter has to be animatable
+			!(x.find("value_desc")->second.get_value_desc()).get_static() &&
```

Comment:
You should check if "value_desc" key exists before using it.

Reply:
hmm yes I was thinking about that but I saw it wasn't checked in other places but yes this is safer will do :).